drp/resource_drp_machine.go

```diff
@@ -364,17 +382,37 @@ func resourceMachineDelete(d *schema.ResourceData, meta interface{}) error {
 	machineObj := obj.(*models.Machine)
 	newObj := models.Clone(machineObj).(*models.Machine)
 
-	if ns, ok := d.GetOk("decommission_stage"); ok {
-		newObj.Stage = ns.(string)
+	if nc, ok := d.GetOk("decommission_color"); ok {
+		newObj.Meta["color"] = nc.(string)
+	} else {
+		newObj.Meta["color"] = "black"
+	}
+
+	if ni, ok := d.GetOk("decommission_icon"); ok {
+		newObj.Meta["icon"] = ni.(string)
 	} else {
-		if machineObj.Stage != "" {
-			newObj.Stage = "discover"
+		newObj.Meta["icon"] = "map outline"
+	}
+
+	if nw, ok := d.GetOk("decommission_workflow"); ok {
+		newObj.Workflow = nw.(string)
+	} else {
+		if ns, ok := d.GetOk("decommission_stage"); ok {
+			newObj.Stage = ns.(string)
 		} else {
-			newObj.BootEnv = "sledgehammer"
+			if machineObj.Workflow != "" {
+				newObj.Workflow = "discover"
+			} else {
+				if machineObj.Stage != "" {
+					newObj.Stage = "discover"
+				} else {
+					newObj.BootEnv = "sledgehammer"
+				}
+			}
 		}
+		// Since we are rebooting, set not runnable.
```

Comment:
Well - Runnable should only be set to false if we aren't using workflows.

Reply:
updating comment